airflow/www/static/js/dag/details/taskInstance/Xcom/XcomEntry.tsx

```diff
@@ -66,6 +67,10 @@ const XcomEntry = ({
         No value found for XCom key
       </Alert>
     );
+  } else {
+    const xcomStr = xcom?.value as string;
+    const xcomStrR = xcomStr.replace(/'/g, '"');
```

Comment:
Can you leave a comment to remind us later what we need this regex for?

Reply:
Also, not a big fan of these two variable names.

How about `const xcomString = xcom?.value ? xcom.value.replace(/'/g, '"') : null;`

Reply:
@bbovenzi  Okay, I am also not happy about the `replace()` and it might be a bit hacky. Do you have an idea to make it "proper"?

Problem is: The serialized dict is serialized in Python in the XCom. The API response with a string containing the serialized XCom from the Python serialization. We get this as a "bad string" in JavaScript.

Best would be to change the API but I'd assume this will be treated as breaking change, getting a "real" object/JSon dict.

Currently the API generates something like:
`{'key': 'value'}`

whereas the JavaScript `JSON.parse()` expects:
`{"key": "value"}`

The regex is fixing the simple things... but how about if the XCom is Python was:
`return {"key": "value", "quote": "\"", "stringify": "'test'", "complex": "\'\"\\"}`

...then the XCom string received is:
`{'key': 'value', 'quote': '\"', 'stringify': \"'test'\", 'complex': '\\'\"\\\\'}`

Do you know (could not find any with Google) proper way to convert Python dict/string to proper JavaScript JSON? All examples propose how to make it "right" in Python.

Reply:
Even after a sleep-over have no better idea than the xcom... left a comment in the code as proposed.
Made conditionals by checking xcom and value in the condition above, then no undefined/conditional regex needed.

Reply:
Extended the API to allow string return or "native" results - which now is consumed by UI to render JSON... WDYT?